Sources/FoundationEssentials/FileManager/FileManager+SymbolicLinks.swift

```diff
@@ -97,6 +121,79 @@ extension _FileManagerImpl {
     }
     
     func destinationOfSymbolicLink(atPath path: String) throws -> String {
+#if os(Windows)
+        return try path.withNTPathRepresentation {
+            var faAttributes: WIN32_FILE_ATTRIBUTE_DATA = .init()
+            guard GetFileAttributesExW($0, GetFileExInfoStandard, &faAttributes) else {
+                throw CocoaError.errorWithFilePath(path, win32: GetLastError(), reading: true)
+            }
+
+            guard faAttributes.dwFileAttributes & FILE_ATTRIBUTE_REPARSE_POINT == FILE_ATTRIBUTE_REPARSE_POINT else {
+                throw CocoaError.errorWithFilePath(path, win32: ERROR_BAD_ARGUMENTS, reading: true)
+            }
+
+            let hFile: HANDLE = CreateFileW($0, GENERIC_READ, FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE, nil, OPEN_EXISTING, FILE_FLAG_OPEN_REPARSE_POINT | FILE_FLAG_BACKUP_SEMANTICS, nil)
+            if hFile == INVALID_HANDLE_VALUE {
+                throw CocoaError.errorWithFilePath(path, win32: GetLastError(), reading: true)
+            }
+            defer { CloseHandle(hFile) }
+
+            let buffer: UnsafeMutableRawBufferPointer =
+                .allocate(byteCount: Int(MAXIMUM_REPARSE_DATA_BUFFER_SIZE), alignment: 8)
+            guard let pBuffer = buffer.baseAddress else {
+                throw CocoaError.errorWithFilePath(path, win32: ERROR_INVALID_DATA, reading: false)
+            }
+
+            var dwBytesWritten: DWORD = 0
+            guard DeviceIoControl(hFile, FSCTL_GET_REPARSE_POINT, nil, 0, pBuffer, DWORD(buffer.count), &dwBytesWritten, nil) else {
+                throw CocoaError.errorWithFilePath(path, win32: GetLastError(), reading: true)
+            }
+            // Ensure that we have enough data.
+            guard dwBytesWritten >= MAXIMUM_REPARSE_DATA_BUFFER_SIZE else {
```

Comment:
Should this be a `guard` that checks if the number of bytes written is less than or equal to the size of the buffer instead of greater than or equal to? If `dwBytesWritten > MAXIMUM_REPARSE_DATA_BUFFER_SIZE` that means our buffer wasn't big enough right?

Reply:
Yeah, I suppose that the right check is `==`. The comment above is because I tripped over this myself for a moment.